app/src/main/java/sdumchykov/androidApp/presentation/viewPager/contacts/MyContactsFragment.kt

```diff
@@ -0,0 +1,302 @@
+        )
+
+        snackbar.setAction("Undo") {
+            parentViewModel.addItem(contact, index)
+
+            Toast.makeText(
+                activity, "${contact.name} has been restored", Toast.LENGTH_LONG
+            ).show()
+        }
+        snackbar.show()
+    }
+
+    private fun createAddContactDialog() {
+        val addContactsBinding = AddContactsDialogBinding.inflate(layoutInflater)
+
+        with(addContactsBinding) {
+            val name = editTextAddContactsDialogName
+            val surname = editTextAddContactsDialogSurname
+            val profession = editTextAddContactsDialogProfession
+
+            dialogAddContact =
+                AlertDialog.Builder(requireActivity()).setView(addContactsBinding.root)
+                    .setOnCancelListener {
+                        clearFieldsInDialog(name, surname, profession)
+                    }.create()
+
+            buttonAddContactsDialogAdd.setOnClickListener {
+                parentViewModel.addNewItem(
+                    name = "${name.text} ${surname.text}", profession = profession.text.toString()
+                )
+
+                clearFieldsInDialog(name, surname, profession)
+                dialogAddContact.dismiss()
+            }
+
+            buttonAddContactsDialogCancel.setOnClickListener {
+                clearFieldsInDialog(name, surname, profession)
+                dialogAddContact.dismiss()
+            }
+
+            binding.textViewMyContactsAddContacts.setOnClickListener { dialogAddContact.show() }
+        }
+    }
+
+    private fun clearFieldsInDialog(
+        name: AppCompatEditText?,
+        surname: AppCompatEditText?,
+        profession: AppCompatEditText?
+    ) {
+        name?.text?.clear()
+        surname?.text?.clear()
+        profession?.text?.clear()
+    }
+
+    private fun handleRecyclerViewContent() {
+        if (parentViewModel.getFetchContactList()) getContactsListWithDexter()
+    }
+
+    private fun getContactsListWithDexter() {
+        Dexter.withActivity(activity).withPermission(Manifest.permission.READ_CONTACTS)
+            .withListener(object : PermissionListener {
+                override fun onPermissionGranted(response: PermissionGrantedResponse) {
+                    if (response.permissionName == Manifest.permission.READ_CONTACTS) {
+                        fetchContacts()
+                    }
+                }
+
+                override fun onPermissionDenied(response: PermissionDeniedResponse) {
+                    Toast.makeText(
+                        activity, "Permission should be granted!", Toast.LENGTH_SHORT
+                    ).show()
+                }
+
+                override fun onPermissionRationaleShouldBeShown(
+                    permission: PermissionRequest,
+                    token: PermissionToken
+                ) {
+                    token.continuePermissionRequest()
+                }
+            }).check()
+    }
+
+    @SuppressLint("Range")
```

Comment:
I suggest you move the whole dexter logic out of the activity to a separated file so as to keep code more readable